server/middlewares/checkUser.js

```diff
@@ -32,18 +31,12 @@ export const checkTeacherPrivilege = (req, res, next) => {
     const { query } = req.params;
     const { role } = req.session;
     if (!['teacher', 'admin', 'super admin'].includes(role)) {
-        return res.status(401).json({
-            status: 'failure',
-            message: 'Sorry, you do not have the required privilege'
-        })
+        return sendError(res, 401, 'Sorry, you do not have the required privilege')
```

Comment:
Kudos to you for introducing `sendError` function. Could you please apply it to all places where it is applicable in the codebase? This will make the codebase looks uniform. Thanks @jamsgra